adapters/appnexus/appnexus.go

```diff
@@ -129,6 +136,41 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.E
 	return requests, append(errs, errors...)
 }
 
+func validateAndBuildAppNexusExt(imp *openrtb2.Imp) (openrtb_ext.ExtImpAppnexus, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return openrtb_ext.ExtImpAppnexus{}, err
+	}
+
+	var appnexusExt openrtb_ext.ExtImpAppnexus
+	if err := json.Unmarshal(bidderExt.Bidder, &appnexusExt); err != nil {
+		return openrtb_ext.ExtImpAppnexus{}, err
+	}
+
+	handleLegacyParams(&appnexusExt)
+
+	if err := validateAppnexusExt(&appnexusExt); err != nil {
```

Comment:
`validateAppnexusExt` only has one validation check. This check can be part of `validateAndBuildAppNexusExt`

```
func validateAppnexusExt(appnexusExt *openrtb_ext.ExtImpAppnexus) error {
	if appnexusExt.PlacementId == 0 && (appnexusExt.InvCode == "" || appnexusExt.Member == "") {
		return &errortypes.BadInput{Message: "No placement or member+invcode provided"}
	}
	return nil
}
```

Reply:
This could be part of `validateAndBuildAppNexusExt`. Any particular reason are you suggesting this change? 
IMO, it is ok to first build the appnexus extension then handle legacy params and then do validation of built appnexus extension in a separate function. This also helps in reducing cognitive load on where are we validating as `validateAppnexusExt` function name makes it clear.

Reply:
We discussed this and we agreed that validateAppnexusExt needs to have its own function makes sense here.